apps/internal/oauth/ops/authority/authority.go

```diff
@@ -332,7 +332,12 @@ func (c Client) AADInstanceDiscovery(ctx context.Context, authorityInfo Info) (I
 		region = detectRegion(ctx)
 	}
 	if region != "" {
-		resp.TenantDiscoveryEndpoint = fmt.Sprintf(TenantDiscoveryEndpointWithRegion, region, authorityInfo.Host, authorityInfo.Tenant)
+		environment := authorityInfo.Host
+		switch environment {
+		case "login.microsoft.com", "login.windows.net", "sts.windows.net", defaultHost:
+			environment = "r." + defaultHost
```

Comment:
tiny perf hint: use `r.login.microsoftonline.com` string constant to prevent concatenation.